specs/networking/libp2p-standardization.md

```diff
@@ -71,14 +98,14 @@ optionally [yamux](https://github.com/hashicorp/yamux/blob/master/spec.md)
 
 ## Gossipsub
 
-#### Protocol id: `/eth/serenity/gossipsub/1.0.0`
+#### Protocol id: `/eth2/beacon_chain/gossipsub/1.0.0/`
```

Comment:
This protocol uses libp2p gossipsub as a pubsub provider. How would a handler for this protocol look like? Would it use gossipsub handler as a delegate?

We might want to explicitly mention gossipsub in the spec, e.g. `/meshsub/1.0.0`. For the sake of conformance with `rpc` I'd suggest to specify beacon chain pubsub in a following way:
- `/eth2/beacon_chain/pubsub/<SchemaVersion>/<Encoding>`
- `/meshsub/1.0.0` is used as underlying protocol

Schema version would handle changes in message format if e.g. an update in aggregation algorithm leads to message patching. If encoding becomes a part of protocol id then topic id could get rid of it which makes forward compatibility less complicated.

There is a question here. What if we want to upgrade to `/meshsub/2.0.0`? What do we do then? But I guess the same question could be addressed to `/mplex` and the others.

Reply:
The current train of thought for putting the encoding inside the topics is so that peers that support different encodings / schemas still connect on the protocol. Peers then in principle could subscribe to two different topics/encodings and act as a relay without having to support two protocols. 

In the event of an upgrade, some nodes could identify other nodes by subscribing to older topics and informing them an upgrade is required. 

In regards to the protocol id. The consensus has been to change the protocol ids to `/eth/..` to avoid accidental connection with ipfs and other non eth-specific nodes. 
If we want to upgrade to a new version of meshsub, we would have to support the new protocol-id. I imagine we would rename the protocol-id to `/eth2/beacon_chain/gossipsub/2.0.0`

In regards to forwards compatibility, I believe the idea would be to coordinate a hard fork such that everyone moves to a new network, either by incrementing the protocol-id version, or in the change of an encoding, a topic prefix. Currently, there are plans to support two encodings. An empty postfix, which has no compression (for interop) then a `snappy` encoding, likely for mainnet which will be realised via two topics, i.e `beacon_block_snappy`.

Reply:
> Peers then in principle could subscribe to two different topics/encodings and act as a relay without having to support two protocols.

IMO, acting as a relay makes things more complicated than supporting two protocols. A relay will have to match different topic ids to a single logical topic and support two encodings. While supporting two protocols only means supporting two encodings. Please, let me know if I am missing something.

A number of topic ids matching to one topic turns into a number of net graphs linked with each other via relays. It requires to have reasonably high number of relays in order to keep not yet upgraded peers subscribed to that topic.